src/components/app-switcher/app-switcher.js

```diff
@@ -0,0 +1,201 @@
+import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * Renders an application drawers that can be used to switch programs
+ * @class AppSwitcher
+ */
+
+export default class AppSwitcher extends React.Component {
+
+  static propTypes = {
+        /**
+         * The json blob for the application links
+         *
+         * @property applicationJson
+         * @type {String}
+         * @default ''
+         */
+    applicationJson: PropTypes.string.isRequired,
+
+        /**
+         * The title of the menu item
+         *
+         * @property menuTitle
+         * @type {String}
+         * @default ''
+         */
+    menuTitle: PropTypes.string
+
+  }
+
+    /**
+     * Return the classes for the styles
+     *
+     * @method styles
+     * @return Hash
+     */
+
+  get styles() {
+    const prefix = 'react-switcher-';
+    return {
+      appItem: prefix + "app-item",
+      menu: classNames((this.state.active ? prefix + "menu-active" : ""), prefix + "menu-title"),
+      sectionTitle: prefix + "section-title",
+      drawer: prefix + "drawer",
+      arrowUp: prefix + "arrow-up",
+      symbol: prefix + "symbol"
+    };
+  }
+
+    /**
+     * Constructor method setting properties
+     *
+     * @method constructor
+     * @return {Void}
+     */
+
+  constructor(...props) {
+    super(...props);
+
+    this.state = {
+      active: false,
+      applications: {}
+    };
+  }
+
+    /**
+     * A lifecycle method that is called after the initial render.
+     *
+     * @method componentDidMount
+     * @return {Void}
+     */
+
+  componentDidMount() {
+    this.setState({applications: JSON.parse(this.props.applicationJson)});
+  }
+
+    /**
+     * A SVG dice icon 16 x 16
+     *
+     * @method SVGIcon
+     * @return JSX Elements
+     */
+
+  SVGIcon = () => {
```

Comment:
Icons should be taken from [this component](https://github.com/Sage/carbon/tree/master/src/components/icon) - if you need some design input for this contact @lukeemmerson